ci-images/stretch/Dockerfile

```diff
@@ -1,8 +1,9 @@
-FROM circleci/node:7
+FROM circleci/buildpack-deps:stretch
 
 USER root
 
 RUN apt-get update && \
+    apt-get upgrade -y && \
```

Comment:
Generally it isn't good practice to do `apt-get upgrade` in containers. If we don't explicitly need this, can we take itout?

Reply:
That's a good point. We definitely don't _need_ that line, and I can remove it. 
 
As I understand it, the reason to avoid `apt-get upgrade` would be to avoid unexpected changes to a Docker image. Basically the same Dockerfile combined with the same code should always return the same result. I can see the reasoning there, but it seems like that's never going to be so straightforward. 

In this case, we're loading from a source image that will continually be updated by CircleCi while still maintaining the same tag, even if that's just to patch some vulnerable OS dependencies. Additionally, on the line below this, we're installing the most recent versions of packages, and that will likely also include unexpected changes over the lifetime of this Dockerfile.

Because we're hosting these images on Quay, the vulnerabilities are quite visible:
<img width="964" alt="screen shot 2018-03-22 at 9 09 05 am" src="https://user-images.githubusercontent.com/419322/37772354-c099e0ac-2db0-11e8-94c8-684a66a8f6a8.png">

The node6 image there is not currently using `apt-get upgrade` while the stretch version is. It definitely adds an additional level of potential change to this image, but from my perspective that _may_ be worth the slightly smaller number of vulnerabilities showing up in Quay.

I'm open to removing this line, just wanted to share my reasoning here.

Reply:
You had me at "patch some vulnerable OS dependencies" 🛡